backend/orchestrators/Orchestrator.py

```diff
@@ -143,8 +143,9 @@ def parse_multi_columns(self, columns: str) -> list:
             return columns.split(",")
 
     # Format request body and headers with relevant info based on search type
-    def prepare_body_headers_with_data(self, request):
+    def prepare_body_headers_with_data(self, request, **kwargs):
         request_messages = request.json["messages"]
+        key=kwargs.get('key', self.AZURE_OPENAI_KEY)
```

Comment:
optional kwargs are passed in here. The key from the environment variable is set as default if none is supplied in the method call.